src/data/mystery-encounters/mystery-encounter-option.ts

```diff
@@ -0,0 +1,260 @@
+  }
+
+  hasRequirements() {
+    return this.requirements.length > 0 || this.primaryPokemonRequirements.length > 0 || this.secondaryPokemonRequirements.length > 0;
+  }
+
+  meetsRequirements(scene: BattleScene) {
+    return !this.requirements.some(requirement => !requirement.meetsRequirement(scene)) &&
+      this.meetsSupportingRequirementAndSupportingPokemonSelected(scene) &&
+      this.meetsPrimaryRequirementAndPrimaryPokemonSelected(scene);
+  }
+
+  pokemonMeetsPrimaryRequirements(scene: BattleScene, pokemon: Pokemon) {
+    return !this.primaryPokemonRequirements.some(req => !req.queryParty(scene.getParty()).map(p => p.id).includes(pokemon.id));
+  }
+
+  meetsPrimaryRequirementAndPrimaryPokemonSelected(scene: BattleScene) {
+    if (!this.primaryPokemonRequirements || this.primaryPokemonRequirements.length === 0) {
+      return true;
+    }
+    let qualified: PlayerPokemon[] = scene.getParty();
+    for (const req of this.primaryPokemonRequirements) {
+      if (req.meetsRequirement(scene)) {
+        if (req instanceof EncounterPokemonRequirement) {
+          const queryParty = req.queryParty(scene.getParty());
+          qualified = qualified.filter(pkmn => queryParty.includes(pkmn));
+        }
+      } else {
+        this.primaryPokemon = undefined;
+        return false;
+      }
+    }
+
+    if (qualified.length === 0) {
+      return false;
+    }
+
+    if (this.excludePrimaryFromSecondaryRequirements && this.secondaryPokemon && this.secondaryPokemon.length > 0) {
+      const truePrimaryPool: PlayerPokemon[] = [];
+      const overlap: PlayerPokemon[] = [];
+      for (const qp of qualified) {
+        if (!this.secondaryPokemon.includes(qp)) {
+          truePrimaryPool.push(qp);
+        } else {
+          overlap.push(qp);
+        }
+
+      }
+      if (truePrimaryPool.length > 0) {
+        // always choose from the non-overlapping pokemon first
+        this.primaryPokemon = truePrimaryPool[Utils.randSeedInt(truePrimaryPool.length, 0)];
+        return true;
+      } else {
+        // if there are multiple overlapping pokemon, we're okay - just choose one and take it out of the supporting pokemon pool
+        if (overlap.length > 1 || (this.secondaryPokemon.length - overlap.length >= 1)) {
+          // is this working?
+          this.primaryPokemon = overlap[Utils.randSeedInt(overlap.length, 0)];
+          this.secondaryPokemon = this.secondaryPokemon.filter((supp) => supp !== this.primaryPokemon);
+          return true;
+        }
+        console.log("Mystery Encounter Edge Case: Requirement not met due to primay pokemon overlapping with support pokemon. There's no valid primary pokemon left.");
+        return false;
+      }
+    } else {
+      // Just pick the first qualifying Pokemon
+      this.primaryPokemon = qualified[0];
+      return true;
+    }
+  }
+
+  meetsSupportingRequirementAndSupportingPokemonSelected(scene: BattleScene) {
```

Comment:
```suggestion
  meetsSupportingRequirementAndSupportingPokemonSelected(scene: BattleScene): boolean {
```
These function names are getting a bit silly lmao